src/newViews/AnalyticalFramework/SecondaryTagging/index.tsx

```diff
@@ -27,20 +39,99 @@ function SecondaryTagging(props: Props) {
 
     // NOTE: intentional console.info
     console.info('secondary tagging in the framework', frameworkId);
+
     const [
         showPreviewModal,
         setShowPreviewModalTrue,
         setShowPreviewModalFalse,
     ] = useModalState(false);
 
+    const [widgets, setWidgets] = useState<Widget[]>([]);
+
+    const [tempWidget, setTempWidget] = useState<PartialWidget | undefined>();
+
+    const handleWidgetAdd = useCallback(
+        (value: PartialWidget) => {
+            setTempWidget(value);
+        },
+        [],
+    );
+
+    const handleWidgetDeleteClick = useCallback(
+        (widgetId: string) => {
+            setWidgets(oldWidgets => deleteWidget(oldWidgets, widgetId));
+        },
+        [],
+    );
+
+    const handleWidgetEditClick = useCallback(
+        (widgetId: string) => {
+            const widget = findWidget(widgets, widgetId);
+            if (widget) {
+                setTempWidget(widget);
+            }
+        },
+        [widgets],
+    );
+
+    const handleWidgetEditCancel = useCallback(
+        () => {
+            setTempWidget(undefined);
+        },
+        [],
+    );
+
+    const handleTempWidgetChange = useCallback(
+        (value: PartialWidget) => {
+            setTempWidget(value);
+        },
+        [],
+    );
+
+    const handleTempWidgetSave = useCallback(
+        (value: Widget) => {
+            setTempWidget(undefined);
+            setWidgets(oldWidgets => injectWidget(oldWidgets, value));
+        },
+        [],
+    );
+
+    const appliedWidgets = useMemo(
+        () => {
+            if (tempWidget) {
+                return injectWidget(widgets, tempWidget);
+            }
+            return widgets;
+        },
+        [tempWidget, widgets],
+    );
+
+    const widgetEditMode = !!tempWidget;
+
+    const editMode = widgetEditMode;
```

Comment:
Lets have only one variable.

Reply:
I know. Doing this so that it's consistent with PrimaryTagging. (Easier to diff them)